nixos/modules/virtualisation/docker-containers.nix

```diff
@@ -185,9 +216,15 @@ let
         ++ [container.image]
         ++ map escapeShellArg container.cmd
       );
-      ExecStartPre = "-${pkgs.docker}/bin/docker rm -f %n";
-      ExecStop = ''${pkgs.bash}/bin/sh -c "[ $SERVICE_RESULT = success ] || ${pkgs.docker}/bin/docker stop %n"'';
-      ExecStopPost = "-${pkgs.docker}/bin/docker rm -f %n";
+
+      ExecStartPre = ["-${pkgs.docker}/bin/docker rm -f ${name}"
+                      "-${pkgs.docker}/bin/docker image prune -f"] ++
+        (if (container.imageFile != null)
+                then ["${pkgs.docker}/bin/docker load -i ${container.imageFile}"]
+                else ["${pkgs.docker}/bin/docker pull ${container.image}"]);
```

Comment:
Why do you want to explicitly pull the image? Since if the image is not locally found, Docker will try to pull it before the image is ran.

Reply:
I think I ran into some weird state issues with failed container units, but I can't remember. I wrote this 7 months ago. I've removed it for now.